### ISSUE TITLE
Cannot start Cadence Server: Got error in github.com/onflow/cadence/languageserver@v0.15.2/server/server.go:507

### ISSUE THREAD

**Issue by Bhawana15** [onflow/cadence#996]
### Problem

While trying to start Cadence server in VS Code, I am getting error as follows:
```
github.com/onflow/cadence/languageserver@v0.15.2/server/server.go:1363 +0x3c5
github.com/onflow/cadence/languageserver/server.(*Server).checkAndPublishDiagnostics(0xc000207170, 0x168a8f8, 0xc00052e5d0, 0xc00004a640, 0x4d, 0xc00026e000, 0x3f0e, 0x3ff0000000000000)
	github.com/onflow/cadence/languageserver@v0.15.2/server/server.go:507 +0xa5
[Error - 4:41:36 pm] Connection to server is erroring. Shutting down server.

```
**Probably, the error has occured in github.com/onflow/cadence/languageserver@v0.15.2/server/server.go:507**

### Context

I am writing smart contracts in Cadence and I am unable to figure out the syntax errors. Also debugging has become very cumbersome as the Cadence server is not running.

**Comment by turbolent** [comment]
Hi @Bhawana15! Thank you for reporting this issue.

- v0.15.2 is not the most recent version, could you please update the CLI and try again?
- Could you please share the code that is causing this crash?

**Comment by Bhawana15** [comment]
Hi @turbolent ! Thanks for your reply.

- I tried updating the Flow CLI, but probably its not working for me. 

- **[Here](https://github.com/Bhawana15/Kitty-Auction-House/blob/main/cadence/contracts/Auction.cdc)** is the code that is causing this crash.

> Hi @Bhawana15! Thank you for reporting this issue.
> 
> * v0.15.2 is not the most recent version, could you please update the CLI and try again?
> * Could you please share the code that is causing this crash?

**Comment by SupunS** [comment]
Thank you for the details.

> * I tried updating the Flow CLI, but probably its not working for me.

Did the CLI update also fail? or the issue didn't resolve even after updating the flow-CLI?

If it is the later, did you also re-install the vs-code-extensions after updating the flow-CLI?

Can you also provide the following info:
- The vs-code version
- OS details/version

**Comment by Bhawana15** [comment]
Hi @SupunS, thanks for your reply.
No, the CLI update did not fail. The issue didn't resolve even after updating the flow-CLI.
Yes, I tried re-installing & reloading the vs-code-extension after updating the flow-CLI.
- The vs-code version is 1.57.0
- OS is Ubuntu 16.04 LTS

After updating the flow-CLI, I am getting a similar error as above:
```
github.com/onflow/cadence/languageserver@v0.15.2/protocol/methods.go:38 +0xd0
github.com/onflow/cadence/languageserver/jsonrpc2.(*handler).Handle(0xc0000ce378, 0x1687080, 0xc0000ca010, 0xc00012b200, 0xc000200f50)
	github.com/onflow/cadence/languageserver@v0.15.2/jsonrpc2/server.go:53 +0x2d0
github.com/sourcegraph/jsonrpc2.(*Conn).readMessages(0xc00012b200, 0x1687080, 0xc0000ca010)
	github.com/sourcegraph/jsonrpc2@v0.0.0-20191222043438-96c4efab7ee2/jsonrpc2.go:522 +0x57e
created by github.com/sourcegraph/jsonrpc2.NewConn
	github.com/sourcegraph/jsonrpc2@v0.0.0-20191222043438-96c4efab7ee2/jsonrpc2.go:334 +0x1d5
[Error - 5:52:25 pm] Connection to server got closed. Server will not be restarted.
```

**Comment by turbolent** [comment]
Thank you again for the details @Bhawana15! 

I debugged your code and fixed the crashers in the latest version in #1013. 

Once merged we can update the language server and release it
